common/testing/taskpoller/taskpoller.go

```diff
@@ -55,17 +58,29 @@ type (
 		*TaskPoller
 		pollActivityTaskRequest *workflowservice.PollActivityTaskQueueRequest
 	}
+	nexusTaskPoller struct {
+		*TaskPoller
+		pollNexusTaskRequest *workflowservice.PollNexusTaskQueueRequest
+	}
 	options struct {
 		tv      *testvars.TestVars
 		timeout time.Duration
 	}
-	optionFunc func(*options)
+	optionFunc           func(*options)
+	TaskCompletedRequest struct {
+		WorkflowTaskCompletedRequest *workflowservice.RespondWorkflowTaskCompletedRequest
+		QueryTaskCompletedRequest    *workflowservice.RespondQueryTaskCompletedRequest
+	}
+	TaskCompletedResponse struct {
+		WorkflowTaskCompletedResponse *workflowservice.RespondWorkflowTaskCompletedResponse
+		QueryTaskCompletedResponse    *workflowservice.RespondQueryTaskCompletedResponse
+	}
```

Comment:
Hm, I'm not sure this is the right direction. I understand that the current API doesn't work for these Queries, but 95% of the time, the task poller will probably only deal with WFTs. Right now, without thinking longer about it, I see three other directions:
1. Create a dedicated, exported function just for queries.
2. Change the return type to any and require type assertions.
3. Hide the fact that there's a "legacy" query (AFAIU) and use the "new" query API; but inside the taskpoller do the right thing auto-magically

Reply:
(1) would be the easiest and clearest; let me know if that's a feasible option. I haven't worked with this Query API before.

Reply:
tagging you @ShahabT here as well since you had the exact concern - The main point seems to be to remove the need for `TaskCompletedResponse/Request` structures. 

In my earlier stab at this, I did not have any of these structures and had made one exported function inside `(*workflowTaskPoller).handleTask` and had changed the return type to `any` (all the places that required it). Not, that in this attempt I also intended on using `Versioning3Suite.pollWftAndHandle` since it served as a nice helper function for all my needs. My train of thought was: 

- That "having type assertions and changing the return type to `nil`" or having these two structures were effectively conveying the same purpose: The response can be one of two types - Query or Workflow
- Having understood this, if I were reading the codebase for the first time, I realized I would appreciate knowing that the handler deals with only Queries and Workflow related tasks (which is made clear by the structure in place since it has only those two fields vs `any` would make it a bit confusing)


Hence, I took the more tedious approach and went ahead with this option - I can certainly revert back if you both feel strongly of not keeping this the way it is but would be curious to hear your thoughts